### PR TITLE
Making output serialization const-correct 

### PR DESCRIPTION
**Pull request by AntonBikineev**

### CODE REVIEW

File: hpx/include/serialization.hpp [hunk 1]
```diff
@@ -9,7 +9,6 @@
 #include <hpx/runtime/serialization/serialization_fwd.hpp>
 #include <hpx/runtime/serialization/serialize.hpp>
 
-#include <hpx/runtime/serialization/allocator.hpp>
```
Comment (sithhell) [inline hpx/include/serialization.hpp]: Why was this removed?

Comment (AntonBikineev) [inline hpx/include/serialization.hpp]: @sithhell it wasn't used (allocator was serialized as empty type). Moreover the code was incorrect as it didn't have split macro

**Comment by sithhell** [comment]
LGTM.
